### PR TITLE
Add a single ci job per group with depwarn=error

### PR DESCRIPTION
**Pull request by lgoettgens**
The idea came first up in https://github.com/oscar-system/Oscar.jl/issues/2693#issuecomment-1682050751, and now with #2913 present, I think this is a good time to talk about this proposal.
And due to #2913 not merged, we have the chance to test this PR.

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -30,6 +30,15 @@ jobs:
         os:
```
Comment (lgoettgens) [inline .github/workflows/CI.yml:30]:
```suggestion
        depwarn: ['']
        os:
```
Is this maybe needed for the new jobs to get created?

**Comment by lgoettgens** [comment]
With this change (if it works), the large part of the testsuite ignores deprecation warnings. These are only catches by the new few jobs.
This helps us to still have the large part of the testsuite working when a dependency deprecated something.

**Comment by thofma** [comment]
OK, I see. Would also be fine with me. Compared to testing deprecations in the downstream tests via OscarDevTools, the solution here is better. What do you think @benlorenz?

**Comment by benlorenz** [comment]
I think separating the deprecations check is better than enabling depwarn for OscarDevTools.

But we need to run doctests with depwarn as well.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (thofma) [reply .github/workflows/CI.yml:30]: You could also just add a separate job, if you don't want to do this dance.

Comment (benlorenz) [reply .github/workflows/CI.yml:30]: > Is this maybe needed for the new jobs to get created?

Yes that seems likely, you can use the `include` block for two things: add extra variables to existing (matching) jobs or to add extra jobs. Which one of these is used depends on whether all variables are already defined in the matrix definition, more details [here](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations).

**Comment by lgoettgens** [comment]
The jobs are now named correctly. But I expected the depwarn=error tests to fail due to #2913 not merged on this branch.
@benlorenz do you have an idea what's going on?

**Comment by benlorenz** [comment]
The tests run on a temporary merge commit with the current master, so this includes the fix now. The test run before the job / doctest adaptions did fail due to the deprecations for julia 1.9: https://github.com/oscar-system/Oscar.jl/actions/runs/6649926182/job/18069184829

Maybe we can use something like `depwarn: 'depwarn'` for the matrix and then in the job:
```
${{ matrix.depwarn == 'depwarn' && 'error' || 'no' }}
```
This way the label for the job would contain `depwarn` instead of just `error`.

And we should probably adjust either the codecov upload count or whether the new jobs do uploads. The codecov page for the current commit shows 6 uploads: https://app.codecov.io/gh/oscar-system/Oscar.jl/commit/ac1a1855de32b2f104bff0f088105292c5e61393.

**Comment by lgoettgens** [comment]
I decided to name it `depwarn=error` to make clear that deprecations actually error and not just warn.

Furthermore, I disabled codecov for these new jobs. If we expect the to fail kind of often, it would be senseless to add the to the codecov config and wait infinitely for the comments. And they shouldn't bring any new covered lines, or at least not a significant amount.

**Comment by benlorenz** [comment]
The run command for the doctests still has an unconditional `--depwarn=error` (currently line 138):
https://github.com/oscar-system/Oscar.jl/pull/2956/files#diff-3ab46ee209a127470fce3c2cf106b1a1dbadbb929a4b5b13656a4bc4ce19c0b8R138

(The `with` block a few lines above is only for running the normal tests on macos, which is done in the doctests job because we only have 5 parallel macos runners)

**Comment by lgoettgens** [comment]
Oh yeah, true. Thanks for spotting! Should be fixed now.

**Review by benlorenz — APPROVED** [review]
thanks!

**Pull request merged**